addons/godot_tours/core/editor_interface_access.gd

```diff
@@ -2,6 +2,8 @@
 ## Extend this script to add support for more areas of the Godot editor or Godot plugins.
 const Utils := preload("utils.gd")
 
+var viewports: Array[Viewport] = []
```

Comment:
Instead of storing an array of viewports/windows in the editor, can't you use Node.get_viewport() in Overlays.gd to create overlays as needed? This would remove the dependency of overlays.gd on editor_interface_access I think.

You can get the viewport any node is in this way, and check if an overlay exists for the window the target control node is part of.